packages/drupal-decoupled/src/editor/config.ts

```diff
@@ -0,0 +1,340 @@
+import { isComposable, isReference, isReferenceRequired } from "./helpers";
+import { Parser, Preset, WithPresetProps, WithOperationProps } from "./parser";
+import { Data, Field, FieldConfig, FieldConfigExtends, FieldExtend, FieldPreset } from "./types";
+
+type CalculatePresetProps = {
+  preset: FieldPreset
+  config: FieldConfig & FieldConfigExtends
+  uiPropName: string
+}
+
+const calculatePreset = ({ preset, config, uiPropName }: CalculatePresetProps): Preset => {
+  const { type, property = '' } = preset
+  let prop = property.replaceAll('{uiPropName}', uiPropName)
+  let mediaFieldName = 'mediaImage'
+
+  if (type === 'mediaImage') {
+    if (config.media?.fieldName) {
+      mediaFieldName = config.media?.fieldName;
+    }
+    prop = property.replaceAll('{mediaFieldName}', mediaFieldName)
+  }
+
+  if (!prop) {
+    return { preset: type, skipOnNull: config.preset?.skipOnNull || false }
+  }
+
+  return {
+    preset: type,
+    property: prop,
+    skipOnNull: config.preset?.skipOnNull || false,
+  }
+}
+
+const getFields = (field: Field) => {
+  if (field.type === 'array') {
+    return field.arrayFields
+  }
+
+  if (field.type === 'object') {
+    return field.objectFields
+  }
+
+  return {}
+}
+
+export type Patch = WithPresetProps | WithOperationProps;
+
+type Config = {
+  component: string
+  fields: Record<string, FieldExtend>
+  defaultProps: Record<string, unknown>
+  // transformers?: Patch[]
```

Comment:
Do we want to keep this commented for a future implementation?

Reply:
🤦 my bad removed